eureka_ml_insights/data_utils/transform.py

```diff
@@ -360,3 +362,28 @@ def transform(self, df: pd.DataFrame, encoding="cl100k_base") -> pd.DataFrame:
             token_count_column = f"{column}_token_count"
             df[token_count_column] = token_count
         return df
+
+@dataclass
+class MajorityVoteTransform:
+    """Applies the majority vote transformation to the specified model output column per ID."""
+    
+    model_output_col: str = "model_output"  # Default column name for model outputs
+    id_col: str = "ID"  # Default column name for IDs
+    majority_vote_col: str = 'majority_vote'
+    
+    def transform(self, df: pd.DataFrame) -> pd.DataFrame:
+        """
+        Transforms the dataframe by calculating the majority vote of 'model_output' per 'ID'.
+        If the 'model_output' is NaN, it will be droped in the majority vote.
```

Comment:
in -> before calulating

Reply:
Sorry what did you mean here?

Reply:
change the words "in the majority vote" to "before calculating the majority vote"

Reply:
fixed